packages/rum-core/test/error-logging/error-logging.spec.js

```diff
@@ -174,44 +174,60 @@ describe('ErrorLogging', function() {
       .then(() => done())
   })
 
-  it('should install onerror and accept ErrorEvents', function(done) {
+  it('should install global listener for error and accept ErrorEvents', function(done) {
     var count = 0
-    var numberOfErrors = 7
+    var numberOfErrors = 4
+    const original = window.addEventListener
     spyOn(apmServer, 'sendErrors').and.callFake(function(errors) {
       expect(errors.length).toBe(numberOfErrors)
       var error = errors[0]
       expect(error.exception.message).toContain(testErrorMessage)
 
       count = count + errors.length
       if (count === numberOfErrors) {
+        window.addEventListener = original
         done()
       }
     })
 
-    window.onerror = null
+    const addedListenerTypes = []
+    window.addEventListener = function(type, listener) {
+      addedListenerTypes.push(type)
+      original.call(null, type, listener)
+    }
     errorLogging.registerGlobalEventListener()
+    expect(addedListenerTypes).toContain('error')
 
-    expect(typeof window.onerror).toBe('function')
-    var apmOnError = window.onerror
+    const filename = 'filename'
+    const lineno = 1
+    const colno = 2
 
     try {
       throw new Error(testErrorMessage)
     } catch (error) {
-      apmOnError(testErrorMessage, 'filename', 1, 2, error)
+      errorLogging.logErrorEvent({
```

Comment:
Maybe we can call the actual listener that we get from the patching `addEventListener`. I know the listener's code is only one line but if we want to change something in the future we can be more confident that it doesn't break the tests.

Reply:
Sure, sounds good.